index.js

```diff
@@ -266,6 +266,32 @@ Peer.prototype.addTrack = function (track, stream) {
   self._needsNegotiation()
 }
 
+/**
+ * Replace a MediaStreamTrack by another in the connection.
+ * @param {MediaStreamTrack} oldTrack
+ * @param {MediaStreamTrack} newTrack
+ * @param {MediaStream} stream
+ */
+Peer.prototype.replaceTrack = async function (oldTrack, newTrack, stream) {
+  var self = this
+
+  self._debug('replaceTrack()')
+
+  var submap = self._senderMap.get(oldTrack)
+  var sender = submap ? submap.get(stream) : null
+  if (!sender) {
+    self.destroy(new Error('Cannot replace track that was never added.'))
+  }
+  self._senderMap.set(newTrack, submap)
+
+  if (sender.replaceTrack != null) {
+    await sender.replaceTrack(newTrack)
+  } else {
+    this.removeTrack(oldTrack, stream)
+    this.addTrack(newTrack, stream)
```

Comment:
I'm not sure about this fallback. It doesn't seem to be equivalent. ie) 'track' events will fire on some browsers, and not on others.

If people want to use browsers that don't support replaceTrack, maybe they should use removeTrack and addTrack explicitly?

Reply:
What do you think @CGNonofr?

Reply:
You're right, and I already changed my PR! Is throwing an error ok for you?

Reply:
We usually use the `self.destroy` method, and it's needed here since senderMap is left in a bad state. I've made the change 👍